internal/controllers/user_controller.go

```diff
@@ -29,204 +29,148 @@ func NewUserController(repo contractions.UserRepository, v validation.Validation
 }
 
 // GetUserByID return user by id
-// @Summary Show a user
-// @Description Get user by id
-// @Produce  json
-// @Security JWT
-// @Param id path int true "User ID"
-// @Success 200 {object} models.User
-// @Header 200 {string} X-AUTH-TOKEN "qwerty"
-// @Failure 500 {object} app.Response
-// @Router /api/v1/users/{id}/ [get]
-func (ctr *UserController) GetUserByID(c *gin.Context) {
+// description: Get user by id
+// example: /api/v1/users/{id}/ [get]
+func (ctr *UserController) GetUserByID(c echo.Context) error {
 	id, err := strconv.ParseInt(c.Param("id"), 10, 64)
 	if err != nil {
-		c.AbortWithStatusJSON(http.StatusBadRequest, err.Error())
-		return
+		return c.String(http.StatusBadRequest, err.Error())
 	}
 
 	user, err := ctr.repo.GetByID(int(id))
 	if err != nil {
-		c.AbortWithStatusJSON(http.StatusInternalServerError, err.Error())
-		return
+		return c.String(http.StatusInternalServerError, err.Error())
 	}
-
-	c.JSON(http.StatusOK, user)
+	return c.JSON(http.StatusOK, user)
 }
 
-// @Summary Authenticate
-// @Description user authorization
-// @Produce  json
-// @Param username query string true "userName"
-// @Param password query string true "password"
-// @Header 200 {string} Access-Control-Allow_origin "*"
-// @Success 200 {object} app.Response
-// @Failure 500 {object} app.Response
-// @Router /api/v1/auth [post]
-func (ctr *UserController) Authenticate(c *gin.Context) {
-	username, _ := c.GetQuery("username")
-	password, _ := c.GetQuery("password")
+// Authenticate @Summary Authenticate
+// description: user authorization
+// example: /api/v1/auth [post]
+func (ctr *UserController) Authenticate(c echo.Context) error {
+	username := c.QueryParam("username")
+	password := c.QueryParam("password")
 	a := models.Auth{Username: username, Password: password}
 	ok, _ := ctr.validation.Valid(&a)
 
 	if !ok {
+		var errors string
 		if ctr.validation.HasErrors() {
-			// maybe c.Error(valid.Errors)
-			app.MarkErrors(ctr.validation.Errors)
+			// maybe e.Error(valid.Errors)
+			errors = app.MarkErrors(ctr.validation.Errors, true)
```

Comment:
Cerate one [multierror](https://github.com/hashicorp/go-multierror) will be better.